src/strapi/revalidate.ts

```diff
@@ -0,0 +1,29 @@
+import type { WebHookHandler, OutputObject } from '../types'
+
+interface StrapiWebhookResponse {
+  event: (typeof ALLOWED_OPERATIONS)[number]
+  model: string
+  uid: string
+  entry: OutputObject
+}
+
+/*
+  Strapi adda a `publishedAt` for each entry. So, we just need to handle the entry.update.
+  As, the field get's updated for every entry.publush/entry.unpublish too.
+*/
+
+const ALLOWED_OPERATIONS: string[] = ['etnry.create', 'entry.update', 'entry.delete']
+
+export const revalidate: WebHookHandler<StrapiWebhookResponse, unknown> = async (request, cb) => {
+  if (process.env?.TELEPORTHQ_ISR_TOKEN !== request.query?.['TELEPORTHQ_ISR_TOKEN']) {
+    return
+  }
+
+  const content = request.body
+  if (ALLOWED_OPERATIONS.includes(content.event) === false) {
+    console.log(`[ON-DEMAND_ISR]: Received an event that is not allowed: ${content.event}`)
+    return
+  }
+
+  cb(content.entry, content.uid)
```

Comment:
is there not normalisation needed for strapi? Don't we have a normalisation function used when fetching data for CMS Items and Lists from Strapi? If yes, we should make sure we are using the same mapping logic in both places.

Reply:
Yeah, in webhook response they are sending directly as `key`, `value` pairs as regular JSON.

Reply:
Why do we have a strapi normalise then just for the normal CMS List and Item fetches https://github.com/teleporthq/teleport-cms-mappers/blob/main/src/strapi/index.ts#L71 and we don't have one here? Is the response format of the entities different?

Reply:
Yes, they are different

When loaded from the API end-point
![Screenshot 2023-07-27 at 16 19 30](https://github.com/teleporthq/teleport-cms-mappers/assets/11075561/24b0522d-3acb-40ed-b940-a0633fcafb63)

Webhook response
<img width="427" alt="Screenshot 2023-07-27 at 16 25 09" src="https://github.com/teleporthq/teleport-cms-mappers/assets/11075561/12e4d68f-8596-4cc8-8776-30b6366747ee">

Reply:
darm... that sucks